src/hal/utils/halcmd_main.c

```diff
@@ -220,7 +220,7 @@ int main(int argc, char **argv)
             fd = fileno(srcfile);
             fcntl(fd, F_SETFD, FD_CLOEXEC);
         } else {
-            halcmd_set_filename("<stdin>");
+            halcmd_set_filename("");
```

Comment:
here you change the filename from <stdin> to the empty string...

Reply:
Because when you are running it from stdin, you know that you are running from stdin. I think there is no reason to display this. If you run a file, it makes quite sense to have the file name and line number.

Reply:
This style is something I'm familiar with from other tools, such as gcc (clang is similar)
```
$ echo 'f' | gcc -Wall -x c - -D'f=if(wrong code)'
<command-line>: error: expected identifier or ‘(’ before ‘if’
<stdin>:1:1: note: in expansion of macro ‘f’
```

The location